test/humanStandardToken.js

```diff
@@ -1,345 +1,186 @@
-    }).catch((err) => { throw new Error(err) })
+    }
   })
 
-  it('events: should fire Approval event properly', function () {
-    var ctr = null
-    return HumanStandardToken.new(10000, 'Simon Bucks', 1, 'SBX', {from: accounts[0]})
-    .then(function (result) {
-      ctr = result
-      return ctr.approve(accounts[1], '2666', {from: accounts[0]})
-    }).then(function (result) {
-      var approvalLog = result.logs.find((element) => {
-        if (element.event.match('Approval')) { return true } else { return false }
-      })
-      assert.strictEqual(approvalLog.args._owner, accounts[0])
-      assert.strictEqual(approvalLog.args._spender, accounts[1])
-      assert.strictEqual(approvalLog.args._value.toString(), '2666')
-    }).catch((err) => { throw new Error(err) })
+  it('transfers: should transfer 10000 to accounts[1] with accounts[0] having 10000', async () => {
+    await HST.transfer(accounts[1], 10000, {from: accounts[0]})
+    const balance = await HST.balanceOf.call(accounts[1])
+    assert.strictEqual(balance.toNumber(), 10000)
+  })
+
+  it('transfers: should fail when trying to transfer 10001 to accounts[1] with accounts[0] having 10000', () => {
+    return expectThrow(HST.transfer.call(accounts[1], 10001, {from: accounts[0]}))
+  })
+
+  it('transfers: should handle zero-transfers normally', async () => {
+    assert(await HST.transfer.call(accounts[1], 0, {from: accounts[0]}), 'zero-transfer has failed')
+  })
+
+  // NOTE: testing uint256 wrapping is impossible in this standard token since you can't supply > 2^256 -1
+  // todo: transfer max amounts
+
+  // APPROVALS
+  it('approvals: msg.sender should approve 100 to accounts[1]', async () => {
+    await HST.approve(accounts[1], 100, {from: accounts[0]})
+    const allowance = await HST.allowance.call(accounts[0], accounts[1])
+    assert.strictEqual(allowance.toNumber(), 100)
+  })
+
+  it('approvals: msg.sender should approve 100 to SampleRecipient and then NOTIFY SampleRecipient. It should succeed.', async () => {
+    let SRS = await SampleRecipientSuccess.new({from: accounts[0]})
+    await HST.approveAndCall(SRS.address, 100, '0x42', {from: accounts[0]})
+    const allowance = await HST.allowance.call(accounts[0], SRS.address)
+    assert.strictEqual(allowance.toNumber(), 100)
+    const value = await SRS.value.call()
+    assert.strictEqual(value.toNumber(), 100)
+  })
+
+  it('approvals: msg.sender should approve 100 to SampleRecipient and then NOTIFY SampleRecipient and throw.', async () => {
+    let SRS = await SampleRecipientThrow.new({from: accounts[0]})
+    return expectThrow(HST.approveAndCall.call(SRS.address, 100, '0x42', {from: accounts[0]}))
+  })
+
+  // bit overkill. But is for testing a bug
+  it('approvals: msg.sender approves accounts[1] of 100 & withdraws 20 once.', async () => {
+    const balance0 = await HST.balanceOf.call(accounts[0])
+    assert.strictEqual(balance0.toNumber(), 10000)
```

Comment:
For readability: please add a blank line below this assert, and do the same for other asserts in this function, or in similar tests.

My thinking is that there are several 'mini tests' within this test, so in such cases, some whitespace will help to  divide them. If there are two asserts in a row, they are probably part of the same 'mini test', so don't divide them up.

make sense?

Reply:
yes!